### DISCUSSION TITLE
Performant gateways

### DISCUSSION THREAD

**Discussion by justinmoon** [fedimint/fedimint#1443] [General]
Eventually users will need a way to intelligently select gateways for payment. Let's use this issue to discuss how to best accomplish this.

**Comment by justinmoon** [comment]
Current lightning smart contracts include node pubkey of the gateway. Federations (or perhaps individual members so we don't need consensus) could keep success rate and latency statistics for each each gateway and share them with clients.

**Comment by okjodom** [comment]
Can gateways self-report performance and other metadata to the clients or to the mint?  I imagine we can plug into the self monitoring capabilities that nodes already do and make that available via an API.
Whoever is querying this information should still distrust it, but use the info as a baseline for monitoring the reputation of a gateway

**Comment by elsirion** [comment]
I think there need to be two compoments:

* Third party (e.g. federation member) reported success statistics
* Self-reported liquidity (if outbound on one GW is less than what we want to pay there's no need to even try)

What we can't do is just use the failure rate observed by the federation since that can be arbitrarily increased by malicious users.
